### DISCUSSION TITLE
Google access token expiration

### DISCUSSION THREAD

**Discussion by sfarkas1988** [supabase/supabase#29578] [Questions]
Hello, I experienced, that my supabase access token is still valid, but the google access token isn't which I received because I use google as auth provider. I would like to make sure that the google token will be refreshable or at least as long available as the supabase token. It's important, because I want to make google API calls with the currently logged in supabase user. But as soon as the google token is expired, the google API calls of course return 401. 
What is the right behaviour here to make sure supabase and google tokens are both valid at the same time?

Thank you in advance.

**Comment by GaryAustin1** [comment]
You need to handle the provider (google) tokens and refreshing on your own.  Supabase  only gives you the token and refresh token (if it used) from the provider when you sign in.  You then need to store it and deal with refreshing as the provider allows.

If you look here https://github.com/supabase/auth-js/issues/131#issuecomment-1838210098 or supabase/auth-js you might find some other user info on doing it, otherwise it will be specific to the Google OAuth API.

**Reply by GaryAustin1** [reply]
https://www.reddit.com/r/Supabase/comments/13txd1n/how_to_authorise_google_auth_for_offline_use_have/

Maybe that will help.  It talks about offline mode and some other info to pass.

**Comment by sfarkas1988** [comment]
Thank you. I was expecting like this. But unfortunately supabase does not provide the refresh token google sends after a successful authentication. I can#t do the refresh on google API side with the stored provider_token.

**Reply by JWPapi** [reply]
The refresh token is only sent when you access for the first time, or when you do prompt: "consent"  I think they pass it through. I’ll come back if they don’t.

**Reply by JWPapi** [reply]
https://github.com/orgs/supabase/discussions/36740#discussioncomment-15230750
